crates/subspace-archiving/tests/integration/reconstructor.rs

```diff
@@ -346,9 +347,13 @@ fn invalid_usage() {
         let result = Reconstructor::new(RECORD_SIZE, SEGMENT_SIZE)
             .unwrap()
             .add_segment(
-                &iter::repeat_with(|| Some(rand::random::<[u8; PIECE_SIZE]>().into()))
-                    .take(PIECES_IN_SEGMENT as usize)
-                    .collect::<Vec<_>>(),
+                &iter::repeat_with(|| {
+                    let mut piece = Piece::default();
+                    thread_rng().fill(*piece.as_mut());
+                    Some(piece)
+                })
+                .take(PIECES_IN_SEGMENT as usize)
+                .collect::<Vec<_>>(),
```

Comment:
I think you can add `Distribution` implementation for `Piece`:
https://docs.rs/rand/latest/rand/distributions/struct.Standard.html#custom-implementations

Reply:
I think we'll then have to add it to regular dependencies and this is only used in tests and doesn't result in a valid piece (this tests checks for invalid segment specifically). `Default` is already producing invalid pieces, I'd rather not introduce more explicit APIs for that.